### ISSUE TITLE
Regression: Base.__@doc__ problem with :toplevel

### ISSUE THREAD

**Issue by cstjean** [timholy/Revise.jl#246]
On Revise 1.0.3, given this package definition:
```julia
module TestPkg

macro toplev(x)
    esc(Expr(:toplevel, quote
             Base.@__doc__ $x
             end))
end

""" Problem """
@toplev struct Foo end

end # module
```

Then:

```julia
julia> using Revise, TestPkg
[ Info: Recompiling stale cache file /home/cst-jean/.julia/compiled/v1.1/TestPkg/9LPSD.ji for TestPkg [9db6ce8c-046d-11e9-3d6b-c5272c17d34a]

julia> 1   # touch src/TestPkg.jl
┌ Warning: expected block expression got $(Expr(:toplevel, quote
│     #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:5 =#
│     begin
│         struct Foo
│             #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:10 =#
│         end
│         (Base.Docs.doc!)(TestPkg, (Base.Docs.Binding)(TestPkg, :Foo), (Base.Docs.docstr)((Core.svec)(" Fine "), (Dict)(:path => "", :linenumber => 0, :module => TestPkg, (Pair)(:fields, (Dict)()))), Union{})
│     end
│ end)) from #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:9 =# Core.:(@doc) " Fine " #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:10 =# @toplev(struct Foo
│             #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:10 =#
│         end)
└ @ Revise ~/.julia/packages/Revise/yp5KG/src/parsing.jl:141
┌ Warning: expected block expression got $(Expr(:toplevel, quote
│     #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:5 =#
│     begin
│         struct Foo
│             #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:10 =#
│         end
│         (Base.Docs.doc!)(TestPkg, (Base.Docs.Binding)(TestPkg, :Foo), (Base.Docs.docstr)((Core.svec)(" Fine "), (Dict)(:path => "", :linenumber => 0, :module => TestPkg, (Pair)(:fields, (Dict)()))), Union{})
│     end
│ end)) from #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:9 =# Core.:(@doc) " Fine " #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:10 =# @toplev(struct Foo
│             #= /home/cst-jean/.julia/dev/TestPkg/src/TestPkg.jl:10 =#
│         end)
└ @ Revise ~/.julia/packages/Revise/yp5KG/src/parsing.jl:141
1
```

It fails on Revise v1.0.3, but worked fine on v0.7.12. I can bisect it if that helps. It's a problem when using QuickTypes.jl, because it has to generate toplevel struct definitions.

**Comment by cstjean** [comment]
I tried bisecting, but I ran into `ERROR: AssertionError: mex.head == :block`, which I had to `bisect skip`. End result was:

```julia
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
09a3d35739e8e7d75e3dc021da1be7b6c361ee7c
72b4597c32adab175cf0aafecb575e53152a9371
c1151992ebc7f0d68a40fb7111c220c134f67939
a25a9d89b9cb276d40966441bc33bcf34c45bb28
eae92013f176b5fb2be41b204b1a9ea41755120b
0071d5f6ffc215b2a62ee5b39656a447ccf63261
b76ae7c840699feebd08182192de2f93de5961e6
d892d77f9804e494fe0042e7916fec6933795ce0
e09be05d5f625032343043f3ca22ebcc492e793a
0aa44233ffb4c8a067d3c35e9abbc56a71765ef4
e57797a1455aecd16b13a74b0ef95b24fcbf385a
45782642ac0576cc7f77e1b9076d979bc9bf29f1
75ee835b7ec941961d71877ba52831c5166a5ff0
7be2cdaab520922d6aff7ad193fc4b018fdad73d
d53db38d03d5e2af6e38c4e4f1443b91ea078593
5b38ceed85ed464090f85fbade35308de4f5c62a
f4dbc9664d06ddf6e3d60f8438f20edbe0c3d332
cfc8fc7bd858760785d1dc5085f5d9077f7194ec
f59b6437fb59bba54f2cffa7060abbc05ac08766
2cfdde2fb758bcaeff60147a1ffe11d92ad71c6b
We cannot bisect more!
```

**Comment by timholy** [comment]
The stacktrace appears to indicate an easy fix,
https://github.com/timholy/Revise.jl/blob/a2c0fd63412cbc9062e640966efb495d79d4eee9/src/parsing.jl#L140

Maybe try generalizing that to allow `:toplevel`?

**Comment by timholy** [comment]
Fixed by #243

**Comment by cstjean** [comment]
Thank you! Sorry that I haven't been able to contribute of late, life has been very busy.
